### PR TITLE
Logan hansen deal cards

### PR DESCRIPTION
**Pull request by Ithilendil**
Finished up the dealCards function and wrote tests for it.

### CODE REVIEW

**Comment by jak103** [comment]
Someone on the actual-uno-deck team needs to either merge or close this. @marwtki @bjsvedin

File: server/uno_test.go [hunk 1]
```diff
@@ -179,3 +179,87 @@ func TestDrawCard(t *testing.T) {
...
+	game.DrawPile = game.DrawPile[:5]
+	database.SaveGame(*game)
+
+	//run deal cards
+	game, err = dealCards(game.ID)
+	//check if the draw pile refreshed
+	assert.Equal(t, 78, len(game.DrawPile))
+
+
+	//database.SaveGame(*game)		//Use this if you make a modification to the game, drawcard will save it for you but need to call whenever you change game state
```
Comment (Aaron-Fine) [inline server/uno_test.go:252]: Do we need this commented out code?

File: server/uno_test.go [hunk 2]
```diff
@@ -179,3 +179,87 @@ func TestDrawCard(t *testing.T) {
...
+}
+
+
+
+
+
```
Comment (Aaron-Fine) [inline server/uno_test.go:258]: Can we reduce the amount of whitespace here?

File: server/uno_test.go [hunk 3]
```diff
@@ -179,3 +179,87 @@ func TestDrawCard(t *testing.T) {
...
+
+//TODO add asserts that it returns the proper game
+//This is something for us to call so that we can check as needed
+func checkGameAgainstDatabase(t *testing.T, game *model.Game) {
+
```
Comment (Aaron-Fine) [inline server/uno_test.go:264]: Please flesh this test out or remove it.

File: server/uno.go [hunk 4]
```diff
@@ -259,7 +259,12 @@ func drawCard(gameID string, playerID string) (*model.Game, error) {
 
 }
 
-func dealCards(gameID string, username string) (*model.Game, error) {
+
+/*This function will:
+		Deal out 7 cards to each player
+		Set the first card for the game to start from
+*/
+func dealCards(gameID string) (*model.Game, error) {  //will need to go into routeHandlers later and remove username string from the parameters being passed in
```
[the remark this answers is not in the document]
Comment (bjsvedin) [reply server/uno.go:267]: There is no need for the player in the function. so it was removed.

File: server/uno_test.go (hunk 1 above)
Comment (bjsvedin) [reply server/uno_test.go:252]: i've removed commented out code

File: server/uno.go [hunk 5]
```diff
@@ -272,23 +277,35 @@ func dealCards(gameID string, username string) (*model.Game, error) {
...
 		for i := 0; i < 7; i++ {
+
+			//grab the top card from the draw pile
 			lastIndex := len(game.DrawPile) - 1
-			card := game.DrawPile[lastIndex]
-			cards = append(cards, card)
-			game.DrawPile = game.DrawPile[:lastIndex]
+			if lastIndex < 0 {
+				//TODO check if something else should be happening, but for now we will just throw an extra deck on the pile
+				game.DrawPile = generateShuffledDeck()
```
[the remark this answers is not in the document]
Comment (bjsvedin) [reply server/uno.go:290]: Yeah that would be best, though the check just before should show that the drawpile was empty, so it didn't hurt anything the way it was.

File: server/uno_test.go (hunk 2 above)
Comment (bjsvedin) [reply server/uno_test.go:258]: done

File: server/uno_test.go (hunk 3 above)
Comment (bjsvedin) [reply server/uno_test.go:264]: done

**Pull request merged**